### DISCUSSION TITLE
Encountering invalid git HEAD (reference 'refs/heads/master' not found) Error with pysr in Python

### DISCUSSION THREAD

**Discussion by spellbookrz** [MilesCranmer/PySR#468] [Q&A] [answered]
Hello, 
I am encountering an issue with the pysr library in Python where it throws an error `invalid git HEAD (reference 'refs/heads/master' not found)` when trying to call Julia code. This issue arises when pysr attempts to add certain Julia packages as dependencies.

Environment:
```
Python Version: 3.9.18
Julia Version: 1.9.4
Operating System: Windows 10
pysr Version: 0.16.3
```
Steps Already Taken:
Updating Julia Package Registry: Ran `Pkg.update()` in Julia REPL, which completed without indicating any changes needed.
Installing Dependencies in Julia: Successfully installed SymbolicRegression and ClusterManagers in Julia using` Pkg.add()`. No changes were indicated, suggesting these packages were already installed and up-to-date.
Ensuring Correct Julia Environment in Python: Tried setting the Julia environment in Python to ensure pysr is using the same environment where the packages were installed.
Reinstalling pysr: Attempted uninstalling and reinstalling pysr in Python to ensure the latest version.
Current Issue:
Despite these steps, when running pysr.install() in Python, the error julia.core.JuliaError: Exception `'invalid git HEAD (reference 'refs/heads/master' not found)'` persists. This issue seems to occur when pysr tries to add its Julia dependencies.

Seeking Assistance:
I am looking for guidance or suggestions to resolve this issue. Thank you in advance for your time and assistance!

**Comment by MilesCranmer** [comment, marked as answer]
Maybe try doing the dependency install manually from the command line? Like this:

```bash
PYSR_VERSION=0.16.3
SRJL_VERSION=0.22.4  # Note this must be updated when PySR version is changed. See `version.py`
julia --project="@pysr-${PYSR_VERSION}" -e "using Pkg; pkg\"add PyCall ClusterManagers SymbolicRegression@${SRJL_VERSION}\""
```

and then just use PySR normally (without doing the `python -m pysr install`, as we did that above)

**Reply by spellbookrz** [reply]
Thank you for your reply! I have done the steps as you said and it can now work normally!
